### PR TITLE
Feature/comparison component

### PR DESCRIPTION
**Pull request by patrickjfl**
New component for data comparison with England & Wales, set up to be flexible for the needs of the other comparisons in the designs.
Screenshot 2022-01-25 at 09 16 07

We have also changed the data handing and cached data structure slightly to serve the component and for improved design and consistency.  Now:
- there is a separate englandAndWalesData store
- all GeodataAPIDataService requests return data in the same 2D Map format
- the fetch request that was being made in the CensusTableByLocation component has now been moved up to the [categorySlug] page. Although not strictly necessary because of the behaviour of stores in Svelte, this is hopefully more intuitive as the page is the parent of CensusTableByLocation and DataComparison, both of which use this data.

The England & Wales geocode has also been moved to config.

### CODE REVIEW

File: src/data/apiMetadata.js [hunk 1]
```diff
@@ -1,3187 +1,3187 @@
 export default [
   {
```
Comment (VivianAllen) [inline src/data/apiMetadata.js]: Why these changes? Is there an issue with ordering?

Comment (patrickjfl) [reply src/data/apiMetadata.js]: think it's just prettier changing the formatting?

File: src/model/censusdata/censusdata.js [hunk 2]
```diff
@@ -37,9 +39,21 @@ export function reset() {
...
-export async function fetchAllDataForGeography(censusDataService, geographyCode) {
+export async function fetchAllDataForGeography(censusDataService, geographyCode, selectedGeography, overwriteCache) {
   const data = await censusDataService.fetchAllDataForGeography(geographyCode);
-  selectedGeographyData.set(data);
+  //if E&W data - call on app initialise
+  if (geographyCode == config.eAndWGeoCode) {
+    englandAndWalesData.set(data);
+  } else if (selectedGeography) {
+    //write data to selectedGeographyData store
+    selectedGeographyData.set(data);
```
Comment (VivianAllen) [inline src/model/censusdata/censusdata.js:49]: what happens if both selectedGeography AND overwriteCache are supplied?

Comment (patrickjfl) [reply src/model/censusdata/censusdata.js:49]: then it will overwrite selectedGeographyData and then exit the if/else conditions and function.

Comment (patrickjfl) [reply src/model/censusdata/censusdata.js:49]: the overwrite happens by default but I think that's fine in this case

Comment (VivianAllen) [reply src/model/censusdata/censusdata.js]: as discussed offline, some of these routes are for behaviour thats not needed yet, so have been left out.

**Review by VivianAllen — APPROVED** [review]
looking good ✅
